### PR TITLE
Minor PIT improvements

### PR DESCRIPTION
**Pull request by ascpixi**
This PR adds minor improvements to the PIT:
- minor formatting changes
- added documentation
- ability to read the CPU state when the timer interrupt has occurred, allowing for things like basic preemptive multitasking
- timeouts changed from signed 32-bit integers to unsigned 64-bit integers, allowing for higher delays between each timer cycle

### CODE REVIEW

**Comment by terminal-cs** [comment]
Could you change all nanosecond fields to ulong instead of int? Int limits the max time to ~2.3 seconds and we don't need values less than 0

**Comment by AsertCreator** [comment]
will pc speaker work?

**Comment by ascpixi** [comment]
> will pc speaker work?

This PR only affects channel 0 of the PIT, so the PC speaker shouldn't be affected, as it runs on channel 2 (I haven't tested it though...)

File: source/Cosmos.HAL2/PIT.cs [hunk 1]
```diff
@@ -1,280 +1,341 @@
...
+            {
+                if (value > 54918330) {
+                    throw new ArgumentException("Delay must be no greater than 54918330");
+                }
+
+                T2Countdown = (ushort)(value / PITDelayNS);
+            }
+        }
+
+        public void EnableSound()
```
Comment (quajak) [inline source/Cosmos.HAL2/PIT.cs:202]: Can we also deprecate all the sound methods?

File: source/Cosmos.HAL2/PIT.cs [hunk 2]
```diff
@@ -1,280 +1,341 @@
...
+                    }
+                    else
+                    {
+                        handler.ID = -1;
+                        activeHandlers.RemoveAt(i);
+                    }
+
+                    handler.HandleTrigger(aContext);
+                } else {
+					handler.NSRemaining -= T0Delay;
```
Comment (quajak) [inline source/Cosmos.HAL2/PIT.cs:288]: Formatting looks off.

**Review by quajak — CHANGES_REQUESTED** [review]
Thanks for the great contribution! Just some minor points

File: source/Cosmos.HAL2/PIT.cs (hunk 1 above)
Comment (ascpixi) [reply source/Cosmos.HAL2/PIT.cs:202]: Sure, will mark with `[Obsolete]` as to not break any old kernels

File: source/Cosmos.HAL2/PIT.cs (hunk 2 above)
Comment (ascpixi) [reply source/Cosmos.HAL2/PIT.cs:288]: woops

**Comment by AsertCreator** [comment]
wait, i noticed `PIT.PITTimer.OnTrigger` does not use reference parameter. therefore i cant, for example, implement premptive multitasking. i can only read registers (maybe)

**Comment by ascpixi** [comment]
> wait, i noticed `PIT.PITTimer.OnTrigger` does not use reference parameter. therefore i cant, for example, implement premptive multitasking. i can only read registers (maybe)

Writing data to the structure won't affect registers... only actually writing to the registers using assembly will. The IRQContext structure is intended to be only read from, CMIIW

**Review by quajak — APPROVED** [review]
Thanks!

**Pull request merged**